### PR TITLE
Adding SimuLink and AutoSAR Metamodels

### PR DESCRIPTION
**Pull request by FabianEger**

### CODE REVIEW

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-SymbolicName: AutoSAR;singleton:=true
+Automatic-Module-Name: AutoSAR
+Bundle-Version: 0.1.0.qualifier
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/META-INF/MANIFEST.MF]:
```suggestion
Bundle-Version: 1.8.0.qualifier
```

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-SymbolicName: AutoSAR;singleton:=true
+Automatic-Module-Name: AutoSAR
+Bundle-Version: 0.1.0.qualifier
+Bundle-ClassPath: .
+Bundle-Vendor: %providerName
+Bundle-Localization: plugin
+Bundle-RequiredExecutionEnvironment: J2SE-1.5
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/META-INF/MANIFEST.MF]:
```suggestion
Bundle-RequiredExecutionEnvironment: JavaSE-17

```

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-SymbolicName: AutoSAR;singleton:=true
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/META-INF/MANIFEST.MF]:
```suggestion
Bundle-SymbolicName: edu.kit.ipd.sdq.metamodels.autosar;singleton:=true

```

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,16 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-SymbolicName: AutoSAR;singleton:=true
+Automatic-Module-Name: AutoSAR
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/META-INF/MANIFEST.MF]:
```suggestion
Automatic-Module-Name: edu.kit.ipd.sdq.metamodels.autosar
```

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/build.properties [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#
+
+bin.includes = .,\
+               model/,\
+               META-INF/,\
+               plugin.xml,\
+               plugin.properties
+jars.compile.order = .
+source.. = src-gen/
+output.. = bin/
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/build.properties]:
```suggestion
output.. = target/classes/
```

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/plugin.properties [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#
+
+pluginName = AutoSAR
+providerName = www.example.org
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/plugin.properties]:
```suggestion
providerName = KIT SDQ
```

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/plugin.properties [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#
+
+pluginName = AutoSAR
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/plugin.properties]:
```suggestion
pluginName = AutoSAR Metamodel
```

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/plugin.xml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+<!--
+-->
+
+<plugin>
+
+   <extension point="org.eclipse.emf.ecore.generated_package">
+      <!-- @generated autoSAR -->
+      <package
+            uri="http://www.example.org/autoSAR"
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/plugin.xml]:
```suggestion
            uri="edu.kit.ipd.sdq.metamodels.autosar"
```

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/plugin.xml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+            class="autoSAR.AutoSARPackage"
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/plugin.xml]:
```suggestion
            class="edu.kit.ipd.sdq.metamodels.autosar.AutoSARPackage"

```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/META-INF/MANIFEST.MF [hunk 10]
```diff
@@ -0,0 +1,16 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-SymbolicName: SimuLink;singleton:=true
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/META-INF/MANIFEST.MF]:
```suggestion
Bundle-SymbolicName: edu.kit.ipd.sdq.metamodels.simulink;singleton:=true
```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/META-INF/MANIFEST.MF [hunk 11]
```diff
@@ -0,0 +1,16 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-SymbolicName: SimuLink;singleton:=true
+Automatic-Module-Name: SimuLink
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/META-INF/MANIFEST.MF]:
```suggestion
Automatic-Module-Name: edu.kit.ipd.sdq.metamodels.simulink
```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/META-INF/MANIFEST.MF [hunk 12]
```diff
@@ -0,0 +1,16 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-SymbolicName: SimuLink;singleton:=true
+Automatic-Module-Name: SimuLink
+Bundle-Version: 0.1.0.qualifier
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/META-INF/MANIFEST.MF]:
```suggestion
Bundle-Version: 1.8.0.qualifier
```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/META-INF/MANIFEST.MF [hunk 13]
```diff
@@ -0,0 +1,16 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: %pluginName
+Bundle-SymbolicName: SimuLink;singleton:=true
+Automatic-Module-Name: SimuLink
+Bundle-Version: 0.1.0.qualifier
+Bundle-ClassPath: .
+Bundle-Vendor: %providerName
+Bundle-Localization: plugin
+Bundle-RequiredExecutionEnvironment: J2SE-1.5
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/META-INF/MANIFEST.MF]:
```suggestion
Bundle-RequiredExecutionEnvironment: JavaSE-17
```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/build.properties [hunk 14]
```diff
@@ -0,0 +1,10 @@
+#
+
+bin.includes = .,\
+               model/,\
+               META-INF/,\
+               plugin.xml,\
+               plugin.properties
+jars.compile.order = .
+source.. = src-gen/
+output.. = bin/
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/build.properties]:
```suggestion
output.. = target/classes/
```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/model/simuLink.aird
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/model/simuLink.aird:1]: .aird files contain the visual representation and can be omitted, so I suggest to also omit them (as they can be generated again if needed).

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/model/autoSAR.aird
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/model/autoSAR.aird:1]: omit, see other comment regarding .aird

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/plugin.properties [hunk 15]
```diff
@@ -0,0 +1,4 @@
+#
+
+pluginName = SimuLink
+providerName = www.example.org
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/plugin.properties]:
```suggestion
providerName = KIT SDQ
```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/plugin.properties [hunk 16]
```diff
@@ -0,0 +1,4 @@
+#
+
+pluginName = SimuLink
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/plugin.properties]:
```suggestion
pluginName = SimuLink Metamodel
```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/plugin.xml [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+
+<!--
+-->
+
+<plugin>
+
+   <extension point="org.eclipse.emf.ecore.generated_package">
+      <!-- @generated simuLink -->
+      <package
+            uri="http://hu.bme.mit.massif/simulink/1.0"
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/plugin.xml]:
```suggestion
            <!-- source uri="http://hu.bme.mit.massif/simulink/1.0" adapted for usage in Vitruvius (vitruv.tools) context -->
            uri="edu.kit.ipd.sdq.metamodels.simulink"
```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/plugin.xml [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+            class="simulink.SimuLinkPackage"
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/plugin.xml]:
```suggestion
            class="edu.kit.ipd.sdq.metamodels.simulink.SimuLinkPackage"

```

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/model/autoSAR.ecore [hunk 19]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ecore:EPackage xmi:version="2.0" xmlns:xmi="http://www.omg.org/XMI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="autoSAR" nsURI="http://www.example.org/autoSAR" nsPrefix="autoSAR">
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.autosar/model/autoSAR.ecore]:
```suggestion
    xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="autoSAR" nsURI="edu.kit.ipd.sdq.metamodels.autoSAR" nsPrefix="autoSAR">
```

File: bundles/edu.kit.ipd.sdq.metamodels.simulink/model/simuLink.ecore [hunk 20]
```diff
@@ -0,0 +1,339 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ecore:EPackage xmi:version="2.0" xmlns:xmi="http://www.omg.org/XMI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="simulink" nsURI="http://hu.bme.mit.massif/simulink/1.0" nsPrefix="simulink">
```
Comment (TomWerm) [inline bundles/edu.kit.ipd.sdq.metamodels.simulink/model/simuLink.ecore]:
```suggestion
    xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="simulink" nsURI="edu.kit.ipd.sdq.metamodels.simulink" nsPrefix="simulink">
```

File: features/edu.kit.ipd.sdq.metamodels.autosar.feature/feature.properties [hunk 21]
```diff
@@ -0,0 +1,131 @@
+featureName=AutoSAR Metamodel
+description=An AutoSAR demo metamodel
+providerName=KIT SDQ
+copyrightURL=https://sdqweb.ipd.kit.edu/
+copyright=\
+Copyright (c) 2014-2020 SDQ\n\
```
Comment (TomWerm) [inline features/edu.kit.ipd.sdq.metamodels.autosar.feature/feature.properties]:
```suggestion
Copyright (c) 2014-2023 SDQ\n\
```

File: features/edu.kit.ipd.sdq.metamodels.autosar.feature/feature.xml [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+   <license url="%licenseURL">
+      %license
+   </license>
+
+   <requires>
+      <import feature="org.eclipse.emf.ecore" version="2.12.0" match="greaterOrEqual"/>
+   </requires>
+
+   <plugin
+         id="edu.kit.ipd.sdq.metamodels.autosar"
```
Comment (TomWerm) [inline features/edu.kit.ipd.sdq.metamodels.autosar.feature/feature.xml:25]: Consistent everywhere AutoSAR or autosar (I prefer the second one)

File: features/edu.kit.ipd.sdq.metamodels.simulink.feature/feature.properties [hunk 23]
```diff
@@ -0,0 +1,131 @@
+featureName=SimuLink Metamodel
+description=An SimuLink demo metamodel
+providerName=KIT SDQ
+copyrightURL=https://sdqweb.ipd.kit.edu/
+copyright=\
+Copyright (c) 2014-2020 SDQ\n\
```
Comment (TomWerm) [inline features/edu.kit.ipd.sdq.metamodels.simulink.feature/feature.properties]:
```suggestion
Copyright (c) 2014-2023 SDQ\n\
```

**Review by TomWerm — CHANGES_REQUESTED** [review]

File: bundles/edu.kit.ipd.sdq.metamodels.autosar/model/autoSAR.ecore (hunk 19 above)
Comment (FabianEger) [reply bundles/edu.kit.ipd.sdq.metamodels.autosar/model/autoSAR.ecore]: without addresses

**Review by FabianEger — COMMENTED** [review]
changed the requested changes from the review

**Review by TomWerm — APPROVED** [review]

**Pull request merged**